src/platform/testing/unit/helpers.js

```diff
@@ -100,11 +100,19 @@ export function mockFetch(returnVal, shouldResolve = true) {
   if (Object.keys(global.fetch).length === 0) {
     oldFetch = global.fetch;
   }
-  global.fetch = sinon
-    .stub()
-    .returns(
-      shouldResolve ? Promise.resolve(returnVal) : Promise.reject(returnVal),
-    );
+
+  global.fetch = sinon.stub().callsFake(url => {
```

Comment:
Whenever you call `mockFetch()`, but don't end up mocking all the calls the stub, an error from the common api helper is returned, because there's no response object. This updates the mock to provide a more typical 404 response.